command/eval_delete.go

```diff
@@ -404,3 +291,40 @@ func correctGrammar(word string, num int) string {
 	}
 	return word
 }
+
+func (e *EvalDeleteCommand) handleDeleteByFilter(filterExpr string) (int, error) {
+
+	// TODO: querying for millions of evals is very expensive... can we add a count RPC?
+
+	// evals, _, err := e.client.Evaluations().List(&api.QueryOptions{
+	// 	Filter: filterExpr,
+	// })
+	// if err != nil {
+	// 	return 1, err
+	// }
+	// evals := []string{}
+
+	// If the user did not wish to bypass the confirmation step, ask this now
+	// and handle the response.
+	// if !e.yes && !e.deleteByArg {
+	// 	code, deleteEvals := e.askQuestion(fmt.Sprintf(
+	// 		"Are you sure you want to delete %v evals? [y/N]",
+	// 		len(evals)), "Cancelling eval deletion")
+	// 	e.Ui.Output("")
+
+	// 	if !deleteEvals {
+	// 		return code, nil
+	// 	}
+	// }
```

Comment:
Open item for discussion: how do we feel about adding a `Eval.Count` RPC that does the exact same thing as `Eval.List` but returns a count instead of serializing potentially millions of evals to JSON?

Reply:
This sounds useful to me.

Reply:
Implemented this in https://github.com/hashicorp/nomad/pull/15117/commits/0ecf61c5d480658e39c5dcd8e11804f4ea408b31 by adding a count field to the `Eval.List` RPC and reusing the same HTTP API with a new shape on `?count=true`. I don't exactly love this approach for the HTTP API, but it seems like it's better than doing `/v1/evaluations/count`?

Reply:
Backed this out in lieu of https://github.com/hashicorp/nomad/pull/15147 and will rebase on that.